api/src/main/java/org/infinispan/api/async/AsyncCache.java

```diff
@@ -0,0 +1,186 @@
+package org.infinispan.api.async;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Flow;
+
+import org.infinispan.api.common.CacheEntry;
+import org.infinispan.api.common.events.cache.CacheContinuousQueryEvent;
+import org.infinispan.api.common.events.cache.CacheEntryEvent;
+import org.infinispan.api.common.events.cache.CacheEntryEventType;
+import org.infinispan.api.common.events.cache.CacheListenerOptions;
+import org.infinispan.api.common.tasks.EntryConsumerTask;
+import org.infinispan.api.configuration.CacheConfiguration;
+
+/**
+ * @since 14.0
+ **/
+public interface AsyncCache<K, V> {
+   /**
+    * The name of the cache.
+    *
+    * @return
+    */
+   String name();
+
+   /**
+    * The configuration for this cache.
+    *
+    * @return
+    */
+   CompletionStage<CacheConfiguration> configuration();
```

Comment:
I might have missed why this needs to be a CompletionStage. Isn't the configuration already defined at this point?